spacy_llm/tasks/ner.py

```diff
@@ -166,21 +168,30 @@ def initialize(
             for initialization.
         nlp (Language): Language instance.
         labels (List[str]): Optional list of labels.
+        infer_prompt_examples (int): How many prompt examples to infer from the Example objects.
+            0 by default. Takes all examples if set to -1.
         """
 
         examples = get_examples()
 
         if not labels:
             labels = list(self._label_dict.values())
+        infer_labels = not labels
 
-        if not labels:
-            label_set = set()
+        if infer_labels:
+            labels = []
 
-            for eg in examples:
+        for eg in examples:
+            if infer_labels:
                 for ent in eg.reference.ents:
-                    label_set.add(ent.label_)
-            labels = list(label_set)
-
+                    labels.append(ent.label_)
+            if (
+                infer_prompt_examples < 0
+                or len(self._prompt_examples) < infer_prompt_examples
+            ):
+                self._prompt_examples.append(self._create_prompt_example(eg))
+
+        labels = list(set(labels))
```

Comment:
Why the list conversion?

Reply:
So in the original version of this code we had a `label_set` variable that was used during label inference, to avoid duplicates, and this was then transformed to a list and put into `labels`. I figured that whichever way the labels come in, we want to avoid having duplicates, and I'm assuming that the list of labels won't be ridiculously long for this operation to become cumbersome.

Reply:
Right, I mean we don't need this to be a `list`? Like
```python
self._label_dict = {self._normalizer(label): label for label in set(labels)}
```
would suffice.

Reply:
Right! I somehow thought that we were storing this list internally. I'll make the change (and add the docs as well).